sources/editor/Stride.GameStudio/Helpers/LanguageInitializer.cs

```diff
@@ -0,0 +1,32 @@
+using System;
+using System.Globalization;
+using Stride.Core.Assets.Editor;
+using Stride.Core.Translation;
+using Stride.Core.Translation.Providers;
+using EditorSettings = Stride.Core.Assets.Editor.Settings.EditorSettings;
+// Copyright (c) .NET Foundation and Contributors (https://dotnetfoundation.org/ & https://stride3d.net) and Silicon Studio Corp. (https://www.siliconstudio.co.jp)
+// Distributed under the MIT license. See the LICENSE.md file in the project root for more information.
+namespace Stride.GameStudio.Helpers
+{
+    internal static class LanguageInitializer
+    {
+        public static void SetLanguage()
+        {
+            TranslationManager.Instance.RegisterProvider(new GettextTranslationProvider());
+            TranslationManager.Instance.CurrentLanguage = EditorSettings.Language.GetValue() switch
+            {
+                SupportedLanguage.MachineDefault => CultureInfo.InstalledUICulture,
+                SupportedLanguage.English => new CultureInfo("en-US"),
+                SupportedLanguage.French => new CultureInfo("fr-FR"),
+                SupportedLanguage.Japanese => new CultureInfo("ja-JP"),
+                SupportedLanguage.Russian => new CultureInfo("ru-RU"),
+                SupportedLanguage.German => new CultureInfo("de-DE"),
+                SupportedLanguage.Spanish => new CultureInfo("es-ES"),
+                SupportedLanguage.ChineseSimplified => new CultureInfo("zh-Hans"),
+                SupportedLanguage.Italian => new CultureInfo("it-IT"),
+                SupportedLanguage.Korean => new CultureInfo("ko-KR"),
+                _ => throw new ArgumentException("Invalid langauge option"),
+            };
+        }
+    }
+}
```

Comment:
A whole file for just one function seems a bit much. I think multiple calls to this function may create side effects, I think it would fit better as a function local to the callsite, that way we don't have any implication that it does.

Reply:
Good point, I've moved method to Program.cs again